src/csharp/Grpc.Core.Api/Metadata.cs

```diff
@@ -343,13 +349,14 @@ public byte[] ValueBytes
 
             /// <summary>
             /// Gets the string value of this metadata entry.
+            /// If the metadata entry is binary then an exception is thrown.
             /// </summary>
             public string Value
             {
                 get
                 {
                     GrpcPreconditions.CheckState(!IsBinary, "Cannot access string value of a binary metadata entry");
-                    return value ?? EncodingASCII.GetString(valueBytes);
```

Comment:
why this change?
at the same time, we're keeping similar logic in ValueBytes, so I'd like to understand the reasoning behind it.

Reply:
It is impossible to get here. If the IsBinary = true (which checks that value is null) then an error is thrown.

Correct me if I'm wrong.

Reply:
I think you're right.